apps/condo/domains/billing/tasks/sendBillingReceiptsAddedNotifications.js

```diff
@@ -0,0 +1,223 @@
+            userId: resident.user.id,
+            url: `${conf.SERVER_URL}/billing/receipts/${billingReceipt.id}`,
+            billingReceiptId: billingReceipt.id,
+            billingAccountId: billingReceipt.account.id,
+            billingPropertyId: billingProperty.id,
+            period: billingReceipt.period,
+        }
+
+        // TODO: calculate message type
+        const messageType = BILLING_RECEIPT_ADDED_TYPE
+        const messageData = {
+            lang: locale,
+            to: { user: { id: resident.user.id } },
+            type: messageType,
+            meta: { dv: 1, data },
+            sender: { dv: 1, fingerprint: 'sendReceiptsAddedNotificationsToPropResidents' },
+        }
+
+        await sendMessage(keystone, messageData)
+    }
+
+    console.info('[INFO] notifications sent:', count)
+    console.info('\r\n\r\n')
+}
+
+const sendReceiptsAddedNotificationsToContextProps = async (contextId, maxSendPerProp) => {
+    const { keystone } = await getSchemaCtx('BillingIntegrationOrganizationContext')
+    const loadByChunks = await getLoadByChunks(keystone)
+    const contextData = await BillingIntegrationOrganizationContext.getOne(keystone, { id: contextId, deletedAt: null })
+
+    if (isEmpty(contextData)) {
+        console.error(`contextId is invalid ${contextId}`)
+
+        return
+    }
+
+    const organization = await Organization.getOne(keystone, { id: contextData.organization.id, deletedAt: null })
+    const locale = get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+
+    const billingProperties = await loadByChunks(BillingProperty, { context: { id: contextId, deletedAt: null }, deletedAt: null }, ['id_ASC'])
+
+    if (isEmpty(billingProperties)) {
+        console.error(`no billing properties found for contextId ${contextId}`)
+
+        return
+    }
+
+    console.info('[INFO] properties found for context:', { contextId, properties: billingProperties.length })
+
+    for (const billingProperty of billingProperties) await sendReceiptsAddedNotificationsToPropResidentsWithPushtokens(billingProperty.id, locale, maxSendPerProp)
+}
+
+const sendBillingReceiptsAddedNotifications = async (maxSendPerProp) => {
+    const { keystone } = await getSchemaCtx('BillingIntegrationOrganizationContext')
+    const loadByChunks = await getLoadByChunks(keystone)
+    const contexts = await loadByChunks(BillingIntegrationOrganizationContext, { status_in: [CONTEXT_FINISHED_STATUS], deletedAt: null }, ['id_ASC'])
+
+    if (isEmpty(contexts)) throw new Error(`No contexts found for status: ${CONTEXT_FINISHED_STATUS}`)
+
+    console.info('[INFO] contexts found: ', contexts.length)
+
+    for (const contextData of contexts) await sendReceiptsAddedNotificationsToContextProps(contextData.id, maxSendPerProp)
+}
+
+
+module.exports = {
+    sendBillingReceiptsAddedNotifications,
+    sendReceiptsAddedNotificationsToContextProps,
+    sendReceiptsAddedNotificationsToPropResidentsWithPushtokens,
```

Comment:
you need to think about the naming it's hard to understand what the functions really do